x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/ClientVersion.java

```diff
@@ -87,7 +89,12 @@ static Manifest getManifest(URL url) throws IOException {
             if (url.getProtocol().equals("jar")) {
                 JarURLConnection jarConn = (JarURLConnection) conn;
                 if (jarConn.getEntryName() == null) { // the URL points to a JAR file
-                    return jarConn.getManifest(); // in case of a fat JAR, this would return the outermost JAR's manifest
+                    JarFile jar = jarConn.getJarFile();
```

Comment:
I suggest using the try/close approach - it's similar but more concise:

```suggestion
if (jarConn.getEntryName() == null) {
 try (JarFile jar = jarconn.getJarFile()) {
      return jar.getManifest();
  }
}
```

Reply:
Beautiful, forgot `JarFile` implements `AutoCloseable`. Thanks!

Reply:
For the record, that will still interrupt the loading if closing fails (while it shouldn't), but OTOH, I can't imagine a case where opening would work, but closing not.